src/Sentry/Internal/Hub.cs

```diff
@@ -379,6 +379,44 @@ private void ApplyTraceContextToEvent(SentryEvent evt, SentryPropagationContext
         evt.DynamicSamplingContext = propagationContext.GetOrCreateDynamicSamplingContext(_options);
     }
 
+    public bool CaptureEnvelope(Envelope envelope) => CurrentClient.CaptureEnvelope(envelope);
+
+    private void AddBreadcrumbForException(SentryEvent evt, Scope scope)
+    {
+        try
+        {
+            if (!IsEnabled || evt.Exception is not { } exception)
+            {
+                return;
+            }
+
+            var exceptionMessage = exception.Message ?? "";
+            var formatted = evt.Message?.Formatted;
+
+            string breadcrumbMessage;
+            Dictionary<string, string>? data = null;
+            if (string.IsNullOrWhiteSpace(formatted))
+            {
+                breadcrumbMessage = exceptionMessage;
+            }
+            else
+            {
+                breadcrumbMessage = formatted;
+                // Exception.Message won't be used as Breadcrumb message
+                // Avoid losing it by adding as data:
+                data = new Dictionary<string, string>
+                {
+                    {"exception_message", exceptionMessage}
+                };
+            }
```

Comment:
Just double-checking: If I've got my log-level set to `Warning` and the SDK creates an event out of a log warning:
- The minimum level is reached and the integration captures an event, skipping the breadcrumb creation
- From the logging integration we get here
- There was no exception - we bail early
- Warnings don't get added as breadcrumbs by neither the logging integration nor the core SDK?

Reply:
Nice catch... have pushed a new commit to fix (the `SentryLogger` and `SentrySink` now **_do_** create breadcrumbs for events, if no exceptions are associated with those events). 

I think that's the correct behaviour since `CaptureEvent` is also used by `CaptureMessage` (and I assume we don't want to automatically add breadcrumbs for calls to `CaptureMessage`):
https://github.com/getsentry/sentry-dotnet/blob/f22c2d479028cc883bf20dc58828a312856bb2dc/src/Sentry/HubExtensions.cs#L221